### ISSUE TITLE
Build fails with "error: attributes cannot be applied to method arguments"

### ISSUE THREAD

**Issue by mad-de** [briansmith/ring#1001]
Building ring 0.16.14 as a dependency (https://github.com/mad-de/IMPP | cargo run --release) results in the following error:

> error: attributes cannot be applied to method arguments
>    --> /home/madde/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.14/src/aead/gcm.rs:296:26
>     |
> 296 | fn detect_implementation(#[allow(unused_variables)] cpu: cpu::Features) -> Implementation {
>     |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^ attributes are not allowed here
>

**Comment by kriegerse** [comment]
facing the same on build of boringtun project (https://github.com/cloudflare/boringtun/). 
```
error: attributes cannot be applied to method arguments
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.14/src/aead/aes.rs:364:26
    |
364 | fn detect_implementation(#[allow(unused_variables)] cpu_features: cpu::Features) -> Implementation {
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^ attributes are not allowed here
error: attributes cannot be applied to method arguments
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.14/src/aead/gcm.rs:296:26
    |
296 | fn detect_implementation(#[allow(unused_variables)] cpu: cpu::Features) -> Implementation {
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^ attributes are not allowed here
error: aborting due to 2 previous errors
error: failed to compile `boringtun v0.3.0`, intermediate artifacts can be found at `/tmp/cargo-install3JvXc7`
Caused by:
  Could not compile `ring`.
```

**Comment by briansmith** [comment]
Apparently I accidentally increased the minimum version of Rust required. If you upgrade the version of Rust you are using then that should resolve it. Otherwise, if somebody posts a PR that works around this before 0.17 work starts (tomorrow?), I'll merge it and release 0.16.5.

**Comment by ctz** [comment]
In case anyone else comes across this: the minimum version is now 1.39; previously I think it was 1.37.

**Comment by briansmith** [comment]
PR #1004 attempts to reset the MSRV back to 1.36. I spent very little time on it and I got stuck on the wasm32-unknown-unknown target requiring 1.37 due to an indirect dependency through wasm-bindgen.

**Comment by briansmith** [comment]
Fixed with PR #1004.
